### PR TITLE
clean up GCP guide, update NVIDIA driver/CUDA installation

### PR DESCRIPTION
**Pull request by mengdong**
Now the official way to install NVIDIA tesla driver should following:
```
wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-ubuntu1804.pin
mv cuda-ubuntu1804.pin /etc/apt/preferences.d/cuda-repository-pin-600
apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/7fa2af80.pub
add-apt-repository "deb http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/ /"
apt-get update
```

### CODE REVIEW

File: getting-started-guides/csp/gcp/gcp.md [hunk 1]
```diff
@@ -28,52 +27,56 @@ There are three steps to run a sample PySpark XGBoost app using Jupyter notebook
...
+* gs://my-bucket/mortgage-small/trainWithEval/test.csv
 
 
-### Step 2. Create a GPU Cluster with pre-installed GPU drivers, Spark RAPIDS libraries, Spark XGBoost libraries and Jupyter notebook 
+### Step 2. Create a GPU Cluster with pre-installed GPU drivers, Spark RAPIDS libraries, Spark XGBoost libraries and Jupyter Notebook 
 
-Using the `gcloud` command creates a new cluster with Rapids Spark GPU initialization action. The following commands will create a new cluster named `<CLUSTER_NAME>` under your project directory. Here we use Ubuntu as our recommended OS for Spark-XGBoost on GCP.
+Using the `gcloud` command creates a new cluster with Rapids Spark GPU initialization action. The following commands 
+will create a new cluster named `<CLUSTER_NAME>` under your project directory. Here we use Ubuntu as our recommended 
+OS for Spark-XGBoost on GCP. Modify the `GCS_BUCKET=my-bucket` line to specify your google GCP bucket name. Also
```
Comment (aroraakshit) [inline getting-started-guides/csp/gcp/gcp.md]: And also modify `INIT_ACTIONS_BUCKET=my-bucket`

File: getting-started-guides/csp/gcp/gcp.md [hunk 2]
```diff
@@ -88,9 +91,10 @@ gcloud beta dataproc clusters create $CLUSTER_NAME  \
     --num-worker-local-ssds 1 \
     --num-workers $NUM_WORKERS \
     --image-version 1.4-ubuntu18 \
-    --bucket $STORAGE_BUCKET \
-    --metadata JUPYTER_PORT=80,INIT_ACTIONS_REPO="gs://$STORAGE_BUCKET",linux-dist="ubuntu" \
-    --initialization-actions gs://$STORAGE_BUCKET/spark-gpu/rapids.sh \
+    --bucket $GCS_BUCKET \
+    --metadata JUPYTER_PORT=8123,INIT_ACTIONS_REPO="gs://$INIT_ACTIONS_BUCKET",linux-dist="ubuntu",
+    GCS_BUCKET="gs://$GCS_BUCKET" \
```
Comment (aroraakshit) [inline getting-started-guides/csp/gcp/gcp.md]: Remove the new line after ```linux-dist="ubuntu",```

File: getting-started-guides/csp/gcp/gcp.md [hunk 3]
```diff
@@ -107,20 +112,21 @@ This will open the Jupyter Notebook. This notebook is running on the “my-gpu-c
...
+Please build the `sample_xgboost_apps jar` with dependencies as specified in the [guide]
+(/getting-started-guides/building-sample-apps/scala.md) and place the jar file (`sample_xgboost_apps-0.1.4-jar-with-dependencies.jar`) under the `gs://$GCS_BUCKET/spark-gpu` folder. To do this you can either drag and drop files from your local machine into the GCP [storage browser](https://console.cloud.google.com/storage/browser/rapidsai-test-1/?project=nv-ai-infra&organizationId=210881545417), or use the [gsutil cp](https://cloud.google.com/storage/docs/gsutil/commands/cp) as shown before to do this from a command line.
```
Comment (aroraakshit) [inline getting-started-guides/csp/gcp/gcp.md]: remove new line after ```[guide]```

File: getting-started-guides/csp/gcp/gcp.md [hunk 4]
```diff
@@ -207,15 +212,15 @@ Submitting a CPU job on this cluster is very similar. Given below is an example
...
+        -maxDepth=8
 ```
 
 ### Step 5. Clean Up
 
 When you're done working on this cluster, don't forget to delete the cluster, using the following command (replacing the highlighted cluster name with yours):
 
-```
+```bash
     gcloud dataproc clusters delete my-gpu-cluster
```
Comment (aroraakshit) [inline getting-started-guides/csp/gcp/gcp.md]: ```gcloud dataproc clusters delete my-gpu-cluster --region=$REGION```

(```WARNING: Dataproc --region flag will become required in January 2020. Please either specify this flag, or set default by running 'gcloud config set dataproc/region <your-default-region>'```)

**Comment by mengdong** [comment]
Should be good now.

File: getting-started-guides/csp/gcp/gcp.md [hunk 5]
```diff
@@ -28,52 +27,56 @@ There are three steps to run a sample PySpark XGBoost app using Jupyter notebook
...
+gsutil cp -r internal gs://$GCS_BUCKET/spark-gpu/
+gsutil cp rapids.sh gs://$GCS_BUCKET/spark-gpu/rapids.sh
 popd
 pushd datasets/
 tar -xvf mortgage-small.tar.gz
-gsutil cp -r mortgage-small/ gs://$STORAGE_BUCKET/
+gsutil cp -r mortgage-small/ gs://$GCS_BUCKET/
 popd
 wget -O cudf-0.9.1-cuda10.jar https://search.maven.org/remotecontent?filepath=ai/rapids/cudf/0.9.1/cudf-0.9.1-cuda10.jar
 wget -O xgboost4j_2.11-1.0.0-Beta2.jar https://search.maven.org/remotecontent?filepath=ai/rapids/xgboost4j_2.11/1.0.0-Beta2/xgboost4j_2.11-1.0.0-Beta2.jar
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/gcp.md]: Need to update to Beta3 @mengdong

File: getting-started-guides/csp/gcp/gcp.md [hunk 6]
```diff
@@ -28,52 +27,56 @@ There are three steps to run a sample PySpark XGBoost app using Jupyter notebook
...
 wget -O xgboost4j-spark_2.11-1.0.0-Beta2.jar https://search.maven.org/remotecontent?filepath=ai/rapids/xgboost4j-spark_2.11/1.0.0-Beta2/xgboost4j-spark_2.11-1.0.0-Beta2.jar
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/gcp.md]: Beta 3 jars

File: getting-started-guides/csp/gcp/gcp.md [hunk 7]
```diff
@@ -28,52 +27,56 @@ There are three steps to run a sample PySpark XGBoost app using Jupyter notebook
...
-gs://my-bucket/spark-gpu/rapids.sh
-gs://my-bucket/spark-gpu/internal/install-gpu-driver-debian.sh
-gs://my-bucket/spark-gpu/internal/install-gpu-driver-ubuntu.sh
-gs://my-bucket/cudf-0.9.1-cuda10.jar
-gs://my-bucket/xgboost4j-spark_2.11-1.0.0-Beta2.jar
-gs://my-bucket/xgboost4j_2.11-1.0.0-Beta2.jar
-gs://my-bucket/mortgage-small/eval/mortgage-small.csv
-gs://my-bucket/mortgage-small/eval/mortgage-small.csv
-gs://my-bucket/mortgage-small/trainWithEval/test.csv
+* gs://my-bucket/spark-gpu/rapids.sh
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/gcp.md]: Beta3

File: getting-started-guides/csp/gcp/gcp.md [hunk 8]
```diff
@@ -88,9 +91,9 @@ gcloud beta dataproc clusters create $CLUSTER_NAME  \
...
     --image-version 1.4-ubuntu18 \
-    --bucket $STORAGE_BUCKET \
-    --metadata JUPYTER_PORT=80,INIT_ACTIONS_REPO="gs://$STORAGE_BUCKET",linux-dist="ubuntu" \
-    --initialization-actions gs://$STORAGE_BUCKET/spark-gpu/rapids.sh \
+    --bucket $GCS_BUCKET \
+    --metadata JUPYTER_PORT=8123,INIT_ACTIONS_REPO="gs://$INIT_ACTIONS_BUCKET",linux-dist="ubuntu",GCS_BUCKET="gs://$GCS_BUCKET" \
+    --initialization-actions gs://$INIT_ACTIONS_BUCKET/spark-gpu/rapids.sh \
     --optional-components=ANACONDA,JUPYTER \
     --subnet=default \
     --properties '^#^spark:spark.dynamicAllocation.enabled=false#spark:spark.shuffle.service.enabled=false#spark:spark.submit.pyFiles=/usr/lib/spark/python/lib/xgboost4j-spark_2.11-1.0.0-Beta2.jar#spark:spark.jars=/usr/lib/spark/jars/xgboost4j-spark_2.11-1.0.0-Beta2.jar,/usr/lib/spark/jars/xgboost4j_2.11-1.0.0-Beta2.jar,/usr/lib/spark/jars/cudf-0.9.1-cuda10.jar' \
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/gcp.md]: Beta3 updates

File: getting-started-guides/csp/gcp/gcp.md [hunk 9]
```diff
@@ -28,52 +27,56 @@ There are three steps to run a sample PySpark XGBoost app using Jupyter notebook
...
-gs://my-bucket/spark-gpu/internal/install-gpu-driver-debian.sh
-gs://my-bucket/spark-gpu/internal/install-gpu-driver-ubuntu.sh
-gs://my-bucket/cudf-0.9.1-cuda10.jar
-gs://my-bucket/xgboost4j-spark_2.11-1.0.0-Beta2.jar
-gs://my-bucket/xgboost4j_2.11-1.0.0-Beta2.jar
-gs://my-bucket/mortgage-small/eval/mortgage-small.csv
-gs://my-bucket/mortgage-small/eval/mortgage-small.csv
-gs://my-bucket/mortgage-small/trainWithEval/test.csv
+* gs://my-bucket/spark-gpu/rapids.sh
+* gs://my-bucket/spark-gpu/internal/install-gpu-driver-debian.sh
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/gcp.md]: Beta3 Updates

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh [hunk 10]
```diff
@@ -24,16 +29,10 @@ if [[ "${ROLE}" != 'Master' ]]; then
...
-  gsutil cp gs://${STORAGE_BUCKET}/xgboost4j-spark_2.11-1.0.0-Beta2.jar /usr/lib/spark/python/lib/
-  gsutil cp gs://${STORAGE_BUCKET}/xgboost4j-spark_2.11-1.0.0-Beta2.jar /usr/lib/spark/jars/
-  gsutil cp gs://${STORAGE_BUCKET}/xgboost4j_2.11-1.0.0-Beta2.jar /usr/lib/spark/jars/
-  gsutil cp gs://${STORAGE_BUCKET}/cudf-0.9.1-cuda10.jar /usr/lib/spark/jars/
-
-  wget "https://rapidsai-data.s3.us-east-2.amazonaws.com/spark/mortgage.zip"
-  unzip "mortgage.zip"
-  hadoop fs -mkdir "/tmp/xgboost4j_spark"
-  hadoop fs -copyFromLocal mortgage "/tmp/xgboost4j_spark/mortgage"
+  gsutil cp ${GCS_BUCKET}/xgboost4j-spark_2.11-1.0.0-Beta2.jar /usr/lib/spark/python/lib/
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: Beta3 Updates

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh [hunk 11]
```diff
@@ -24,16 +29,10 @@ if [[ "${ROLE}" != 'Master' ]]; then
...
+  gsutil cp ${GCS_BUCKET}/xgboost4j-spark_2.11-1.0.0-Beta2.jar /usr/lib/spark/jars/
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: Beta3 Updates

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh [hunk 12]
```diff
@@ -24,16 +29,10 @@ if [[ "${ROLE}" != 'Master' ]]; then
...
+  gsutil cp ${GCS_BUCKET}/xgboost4j_2.11-1.0.0-Beta2.jar /usr/lib/spark/jars/
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: Beta3

File: getting-started-guides/csp/gcp/gcp.md [hunk 13]
```diff
@@ -28,52 +27,56 @@ There are three steps to run a sample PySpark XGBoost app using Jupyter notebook
...
-gsutil cp rapids.sh gs://$STORAGE_BUCKET/spark-gpu/rapids.sh
+gsutil cp -r internal gs://$GCS_BUCKET/spark-gpu/
+gsutil cp rapids.sh gs://$GCS_BUCKET/spark-gpu/rapids.sh
 popd
 pushd datasets/
 tar -xvf mortgage-small.tar.gz
-gsutil cp -r mortgage-small/ gs://$STORAGE_BUCKET/
+gsutil cp -r mortgage-small/ gs://$GCS_BUCKET/
 popd
 wget -O cudf-0.9.1-cuda10.jar https://search.maven.org/remotecontent?filepath=ai/rapids/cudf/0.9.1/cudf-0.9.1-cuda10.jar
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/gcp.md]: cudf-0.9.1-cuda10.jar should be cudf-0.9.2-cuda10.jar

File: getting-started-guides/csp/gcp/gcp.md (hunk 7 above)
Comment (krajendrannv) [reply getting-started-guides/csp/gcp/gcp.md]: cudf-0.9.1-cuda10.jar should be cudf-0.9.2-cuda10.jar

File: getting-started-guides/csp/gcp/gcp.md (hunk 9 above)
Comment (krajendrannv) [reply getting-started-guides/csp/gcp/gcp.md]: cudf-0.9.1-cuda10.jar should be cudf-0.9.2-cuda10.jar

File: getting-started-guides/csp/gcp/gcp.md (hunk 8 above)
Comment (krajendrannv) [reply getting-started-guides/csp/gcp/gcp.md]: cudf-0.9.1-cuda10.jar should be cudf-0.9.2-cuda10.jar

File: getting-started-guides/csp/gcp/spark-gpu/README.md [hunk 14]
```diff
@@ -30,34 +30,38 @@ all prerequisites installed:
...
-        --optional-components=ZEPPELIN \
+        --bucket $GCS_BUCKET \
+        --metadata JUPYTER_PORT=8123,INIT_ACTIONS_REPO="gs://$INIT_ACTIONS_BUCKET",linux-dist="ubuntu",
+        GCS_BUCKET="gs://$GCS_BUCKET" \
+        --initialization-actions gs://$INIT_ACTIONS_BUCKET/spark-gpu/rapids.sh \
+        --optional-components=ANACONDA,JUPYTER \
         --subnet=default \
-        --properties 'spark:spark.dynamicAllocation.enabled=false,spark:spark.shuffle.service.enabled=false' \
-        --enable-component-gateway       
+        --properties '^#^spark:spark.dynamicAllocation.enabled=false#spark:spark.shuffle.service.enabled=false#spark:spark.submit.pyFiles=/usr/lib/spark/python/lib/xgboost4j-spark_2.11-1.0.0-Beta2.jar#spark:spark.jars=/usr/lib/spark/jars/xgboost4j-spark_2.11-1.0.0-Beta2.jar,/usr/lib/spark/jars/xgboost4j_2.11-1.0.0-Beta2.jar,/usr/lib/spark/jars/cudf-0.9.1-cuda10.jar' \
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/spark-gpu/README.md]: Beta3 and cudf-0.9.1-cuda10.jar should be cudf-0.9.2-cuda10.jar

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh [hunk 15]
```diff
@@ -24,16 +29,10 @@ if [[ "${ROLE}" != 'Master' ]]; then
...
+  gsutil cp ${GCS_BUCKET}/cudf-0.9.1-cuda10.jar /usr/lib/spark/jars/
```
Comment (krajendrannv) [inline getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: cudf-0.9.1-cuda10.jar should be cudf-0.9.2-cuda10.jar

File: getting-started-guides/csp/gcp/gcp.md (hunk 5 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/gcp.md]: done, also need to change `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/gcp.md (hunk 6 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/gcp.md]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/gcp.md (hunk 8 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/gcp.md]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh (hunk 11 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/gcp.md (hunk 13 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/gcp.md]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh (hunk 15 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/spark-gpu/README.md (hunk 14 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/spark-gpu/README.md]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh (hunk 12 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/gcp.md (hunk 9 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/gcp.md]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/gcp.md (hunk 7 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/gcp.md]: done, also need to change maven url and from `xgboost4j_2.11` to `xgboost4j_2.x`

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh [hunk 16]
```diff
@@ -5,9 +5,22 @@ set -euxo pipefail
 readonly ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
 readonly LINUX_DIST=$(/usr/share/google/get_metadata_value attributes/linux-dist)
 
+readonly DEAFULT_INIT_ACTIONS_REPO=gs://dataproc-initialization-actions
 readonly INIT_ACTIONS_REPO="$(/usr/share/google/get_metadata_value attributes/INIT_ACTIONS_REPO ||
   echo ${DEAFULT_INIT_ACTIONS_REPO})"
 
+readonly DEAFULT_GCS_BUCKET=gs://my-bucket
```
Comment (GaryShen2008) [inline getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: Typo: DEAFULT_*
Should be DEFAULT_*?

File: getting-started-guides/csp/gcp/spark-gpu/README.md [hunk 17]
```diff
@@ -11,56 +11,99 @@ Prerequisites
...
+* gs://my-bucket/mortgage-small/trainWithEval/test.csv
+
+
+### Step 2 Using the `gcloud` command to create a new cluster with this initialization action. 
+
+The following command will create a new cluster named `<CLUSTER_NAME>`. Before the init script fully merged into 
+`<dataproc-initialization-actions>` bucket, user need to copy the sparkgpu initialization script into a accessible GCS and into following structure. Ubuntu is recommended as CUDA support ubuntu, debian could be used by modifying `image-version` and `linux-dist` accordingly.
 
+    ```bash
+    /$GCS_BUCKET/sparkgpu/rapids.sh
```
Comment (GaryShen2008) [inline getting-started-guides/csp/gcp/spark-gpu/README.md]: Is "sparkgpu" correct?
Some places above use "spark-gpu".

File: getting-started-guides/csp/gcp/spark-gpu/rapids.sh (hunk 16 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/spark-gpu/rapids.sh]: Good catch, I have fixed it.

File: getting-started-guides/csp/gcp/spark-gpu/README.md (hunk 17 above)
Comment (mengdong) [reply getting-started-guides/csp/gcp/spark-gpu/README.md]: Good catch, I have fixed it.

**Pull request merged**